lib/bus.js

```diff
@@ -1,356 +1,368 @@
-           msg.path = '/org/freedesktop/DBus';
-       if (!msg.destination)
-           msg.destination = 'org.freedesktop.DBus';
-       if (!msg['interface'])
-           msg['interface'] = 'org.freedesktop.DBus';
-       self.invoke(msg, callback);
-    };
+  this.invoke = function (msg, callback) {
+    if (!msg.type) {
+      msg.type = constants.messageType.methodCall;
+    }
+    msg.serial = self.serial;
+    self.serial++;
+    this.cookies[msg.serial] = callback;
+    self.connection.message(msg);
+  };
+
+  this.invokeDbus = function (msg, callback) {
+    if (!msg.path) {
+      msg.path = '/org/freedesktop/DBus';
+    }
+    if (!msg.destination) {
+      msg.destination = 'org.freedesktop.DBus';
+    }
+    if (!msg['interface']) {
+      msg['interface'] = 'org.freedesktop.DBus';
+    }
+    self.invoke(msg, callback);
+  };
 
-    this.mangle = function(path, iface, member) {
-        var obj = {};
-        if (typeof path === 'object') // handle one argumant case mangle(msg)
+  this.mangle = function (path, iface, member) {
+    var obj = {};
+    if (typeof path === 'object') // handle one argumant case mangle(msg)
         {
-            obj.path = path.path;
-            obj['interface'] = path['interface'];
-            obj.member = path.member;
-        } else {
-            obj.path = path;
-            obj['interface'] = iface;
-            obj.member = member;
-        }
-        return JSON.stringify(obj);
+      obj.path = path.path;
+      obj['interface'] = path['interface'];
+      obj.member = path.member;
+    } else {
+      obj.path = path;
+      obj['interface'] = iface;
+      obj.member = member;
+    }
+    return JSON.stringify(obj);
+  };
+
+  this.sendSignal = function (path, iface, name, signature, args) {
+    var signalMsg = {
+      type: constants.messageType.signal,
+      serial: self.serial,
+      'interface': iface,
+      path: path,
+      member: name
     };
-
-    this.sendSignal = function(path, iface, name, signature, args) {
-        var signalMsg = {
-            type: constants.messageType.signal,
-            serial: self.serial,
-            'interface': iface,
-            path: path,
-            member: name
-        };
-        if (signature) {
-            signalMsg.signature = signature;
-            signalMsg.body = args;
-        }
-        self.connection.message(signalMsg);
+    if (signature) {
+      signalMsg.signature = signature;
+      signalMsg.body = args;
     }
+    self.connection.message(signalMsg);
+  };
+
+  this.sendError = function (msg, errorName, errorText) {
+    var reply = {
+      type: constants.messageType.error,
+      replySerial: msg.serial,
+      destination: msg.sender,
+      errorName: errorName,
+      signature: 's',
+      body: [errorText]
+    };
+        // console.log('SEND ERROR', reply);
+    this.connection.message(reply);
+  };
+
+  this.sendReply = function (msg, signature, body) {
+    // TODO (eslint): use reply?
```

Comment:
oh, this was actually meant to be "reply to this call with with body/signature"  helper

could you add commit to fix to be `mesage(reply)` ?